### DISCUSSION TITLE
Inter-process ResourceLocks

### DISCUSSION THREAD

**Discussion by roded** [junit-team/junit-framework#3395] [Q&A]
Hi,
I'm interested in the possibility of extending JUnit's `ResourceLock` mechanism not only for locking inside a single JVM process but for inter-process locking (we might have multiple JVM processes running JUnit tests on the same system with shared resources).

Is it currently possible to replace JUnit's `LockManager` implementation? 
And even if it is possible, is this a good idea?

Thanks

**Comment by dhoard** [comment]
Looking at the code, it's currently not possible to replace it.

https://github.com/junit-team/junit5/blob/22a64e7aad5ae0e3c9cf2a692f77bd3de8d1bbed/junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/NodeTreeWalker.java#L34

If you need complex locking, I suggest that you write your own distributed lock manager (DLM) and use it in your tests either directly (programmatically) or via a custom extension.

**Reply by roded** [reply]
Thanks for the reply @dhoard.
I would much rather use JUnit constructs for synchronization.

I wonder if the JUnit developers consider adding a feature allowing replacement of the `LockManager` to serve such a purpose.
